### PR TITLE
Extract OSM POIs and export to .poi files

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: osm-utils/src/poi.rs [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    fn convert_id(&mut self) {
+        for poi_type in &mut self.poi_types {
+            poi_type.id = format!("poi_type:{}", poi_type.id);
+        }
+        for rule in &mut self.rules {
+            rule.poi_type_id = format!("poi_type:{}", rule.poi_type_id);
+        }
+    }
+}
+const DEFAULT_JSON_POI_TYPES: &str = r#"
```
Comment (nberard) [inline osm-utils/src/poi.rs]: could load from a real json file

File: osm-utils/src/poi.rs [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+        poi_type: poi_type.clone(),
+        properties: make_properties(osmobj.tags()),
+    })
+}
+
+fn format_poi_id(osm_type: &str, id: i64) -> String {
+    format!("poi:osm:{}:{}", osm_type, id)
+}
+
+pub fn pois(pbf: &mut OsmPbfReader, matcher: &PoiConfig) -> Vec<objects::Poi> {
```
Comment (nberard) [inline osm-utils/src/poi.rs]: we could add a verb like extract or something :D

File: osm-utils/src/poi.rs [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+            return None;
+        }
+    };
+    let (id, coord) = match *osmobj {
+        osmpbfreader::OsmObj::Node(ref node) => (
+            format_poi_id("node", node.id.0),
+            objects::Coord::new(node.lon(), node.lat()),
+        ),
+        osmpbfreader::OsmObj::Way(ref way) => {
+            (format_poi_id("way", way.id.0), get_way_coord(obj_map, way))
```
Comment (nberard) [inline osm-utils/src/poi.rs]: @prhod do we need the computed coordinates of a way to be on the original line?

File: osm-utils/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+pub fn make_centroid(boundary: &Option<MultiPolygon<f64>>) -> objects::Coord {
+    let coord = boundary
+        .as_ref()
+        .and_then(|b| b.centroid().map(|c| objects::Coord::new(c.x(), c.y())))
+        .unwrap_or_else(objects::Coord::default);
+    if coord.is_valid() {
+        coord
+    } else {
+        objects::Coord::default()
```
Comment (nberard) [inline osm-utils/src/lib.rs]: default is used twice as an error value, we could return a Result instead

File: src/bin/extract-osm-pois.rs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        Some(path) => {
+            let r = std::fs::File::open(&path)
+                .with_context(|_| format!("Error while opening configuration file {:?}", path))?;
+            PoiConfig::from_reader(r).unwrap()
+        }
+    };
+    let mut osm_reader = make_osm_reader(&opt.input)?;
+
+    info!("Extracting pois from osm");
+    let pois = pois(&mut osm_reader, &matcher);
```
Comment (nberard) [inline src/bin/extract-osm-pois.rs]: using read and write could be ok here

Comment (datanel) [reply src/bin/extract-osm-pois.rs]: I don't read pois I extract pois from an osm pbf.
and write what?

File: osm-utils/src/poi.rs (hunk 1 above)
Comment (datanel) [reply osm-utils/src/poi.rs]: what's the need?

File: src/bin/extract-osm-pois.rs (hunk 5 above)
Comment (datanel) [reply src/bin/extract-osm-pois.rs]: I chose `extract_pois` and `export_pois`

File: Cargo.toml [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
+authors = ["David Quintanel <david.quintanel@gmail.com>", "Nicolas Bérard <nicolas.berard@kisio.org>"]
+edition="2018"
 
 [dependencies]
 navitia_model = { git = "https://github.com/CanalTP/navitia_model" }
 structopt = "0.2.10"
-env_logger = "0.5.13"
-log = "0.4.5"
+env_logger = "0.6.0"
+log = "0.4.6"
```
Comment (kinnou02) [inline Cargo.toml]: it's outside the scope of this PR but I would recommend to use [slog](https://github.com/slog-rs/slog) for logging, it's a pain to setup a first, but will allow use json log and an elk stack easily later.

File: osm-utils/src/poi.rs (hunk 1 above)
Comment (kinnou02) [reply osm-utils/src/poi.rs]: easier to edit, and a file less bloated :p 
it's easy to do with [include_str!](https://doc.rust-lang.org/std/macro.include_str.html)

File: src/bin/extract-osm-pois.rs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+};
+use osm_utils::poi::PoiConfig;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use zip;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "extract_osm_pois",
+    about = "Extract POIs from OSM.",
```
Comment (TeXitoi) [inline src/bin/extract-osm-pois.rs]: you can about in doc comment if you want

File: src/bin/extract-osm-pois.rs [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+fn run() -> Result<()> {
+    info!("Launching extract_osm_pois.");
+    let opt = Opt::from_args();
+    let matcher = match opt.poi_config {
+        None => PoiConfig::default(),
+        Some(path) => {
+            let r = std::fs::File::open(&path)
+                .with_context(|_| format!("Error while opening configuration file {:?}", path))?;
+            PoiConfig::from_reader(r).unwrap()
```
Comment (TeXitoi) [inline src/bin/extract-osm-pois.rs]: you want to `?` not `.unwrap()`

File: Cargo.toml [hunk 9]
```diff
@@ -17,5 +17,6 @@ csv = "1.0.2"
 zip = "0.5.0"
 serde = "1.0.84"
 serde_derive = "1.0.84"
+osmpbfreader = "0.11"
```
Comment (TeXitoi) [inline Cargo.toml]: The last one is 0.12, nothing really interesting appart internal lib update

**Pull request merged**